pkg/destroy/gcp/filestore.go

```diff
@@ -0,0 +1,96 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/api/file/v1"
+
+	gcpconsts "github.com/openshift/installer/pkg/constants/gcp"
+)
+
+func (o *ClusterUninstaller) filestoreParentPath() string {
+	return fmt.Sprintf("projects/%s/locations/-", o.ProjectID)
+}
+
+func (o *ClusterUninstaller) clusterFilestoreLabelFilter() string {
+	return fmt.Sprintf("labels.%s = \"owned\"", fmt.Sprintf(gcpconsts.ClusterIDLabelFmt, o.ClusterID))
+}
+
+func (o *ClusterUninstaller) listFilestores(ctx context.Context) ([]cloudResource, error) {
+	o.Logger.Debug("Listing filestores")
+	result := []cloudResource{}
+
+	ctx, cancel := context.WithTimeout(ctx, defaultTimeout)
+	defer cancel()
+	instSvc := file.NewProjectsLocationsInstancesService(o.fileSvc)
+	lCall := instSvc.List(o.filestoreParentPath()).Context(ctx).Filter(o.clusterFilestoreLabelFilter())
+
+	nextPageToken := "pageToken"
+	for nextPageToken != "" {
+		instances, err := lCall.Do()
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving filestore instances: %w", err)
+		}
+
+		for _, activeInstance := range instances.Instances {
+			if len(activeInstance.FileShares) == 0 {
+				// skip multi-share instances
+				o.Logger.Debugf("Skipping multi-share filestore %s", activeInstance.Name)
+				continue
+			}
+			o.Logger.Debugf("Found filestore %s", activeInstance.Name)
+			result = append(result, cloudResource{
+				name:     activeInstance.Name,
+				typeName: "filestore",
+			})
+		}
+
+		nextPageToken = instances.NextPageToken
+		lCall.PageToken(nextPageToken)
+	}
+	return result, nil
+}
+
+func (o *ClusterUninstaller) deleteFilestore(ctx context.Context, item cloudResource) error {
+	o.Logger.Debugf("Deleting filestore %s", item.name)
+
+	ctx, cancel := context.WithTimeout(ctx, defaultTimeout)
+	defer cancel()
+	instSvc := file.NewProjectsLocationsInstancesService(o.fileSvc)
+	_, err := instSvc.Delete(item.name).Context(ctx).Do()
+	if err != nil && isForbidden(err) {
+		o.deletePendingItems(item.typeName, []cloudResource{item})
+		return fmt.Errorf("Insufficient permissions to delete filestore %s: giving up", item.name)
```

Comment:
the linter is complaining about the caps. If we're not failing on missing perms, should this be an error, or should this be a logrus Warning? (in which case, with logrus warning it should stay capitalized).

Reply:
I'm unsure myself about this part. The uninstallation can't fail as such, so we only need some way to inform the user the resource deletion failed. Perhaps logging a warning and returning an error would work.

Reply:
as the error doesn't stop the destroyer, this is fine. just drop the caps to appease the linter

```suggestion
		return fmt.Errorf("insufficient permissions to delete filestore %s: giving up", item.name)
```

There was one other failure in go-lint that needed to be resolved

Reply:
I have added a log message (so the user is warned when the deletion fails because of missing permissions) and changed the character case for the error.

The other linter failure was caused by the new `isForbidden` function that now takes a different error type so copying the `isNoOp` was apparently incorrect.